bin/run_docker.py

```diff
@@ -267,8 +268,55 @@ def mount_volumes() -> dict:
     return volumes
 
 
+def monitor_container(
+    container: docker.models.containers.Container,
+    timeout: Union[int, float],
+    poll_interval: Union[int, float],
+    elapsed_time: int = 0,
+):
+    """
+    Recursively monitor the status of a Docker container until it finishes,
+    or the timeout is reached, at which point the container is killed.
+
+    Arguments:
+        container: The Docker container subject to monitor.
+        timeout: Maximum duration to monitor the container (in minutes) before forced shutdown.
+        poll_interval: Time to wait between status checks (in minutes).
+        elapsed_time: The elapsed time already spent monitoring (in minutes).
+    """
+
+    # Check if the elapsed time has reached or exceeded the timeout
+    if elapsed_time >= timeout:
+        print("Timeout reached. Stopping container.")
+        container.stop(timeout=10)
+        return f"Container exceeded execution time limit of {timeout} minutes. Unable to process."
+
+    # Refresh container status
+    container.reload()
+
+    # If the container has exited, stop monitoring
+    if container.status == "exited":
```

Comment:
I would take a look at this logic a bit.  There is a scenario where the container finishes at the last cycle, but then it it'll be  "elapsed_time >= timeout" causing the error to potentially be raised.

Reply:
Yeah. If I understand `conainter.reload()` should probably be the first thing this does. Thinking:

1. reload
2. Check if status `exited`
3. Check if exceeding timeout